### PR TITLE
Disable resolving discriminators for semantic validation of a 2.0 openapi spec.

### PR DESCRIPTION
**Pull request by amarzavery**

### CODE REVIEW

**Comment by veronicagg** [comment]
Issue found via https://github.com/Azure/azure-rest-api-specs/pull/1903

File: ChangeLog.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+### 10/24/2017 0.4.17
+- Disable resolving discriminators while performing semantic validation for an open api specification that confirms to 2.0 version.
```
Comment (veronicagg) [inline ChangeLog.md]: conforms?

File: lib/validators/specValidator.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ class SpecValidator {
    * 
    * @param {object} [options.shouldResolveAllOf] Should allOf references be resolved? Default: true
    * 
+   * @param {object} [options.shouldResolveDiscriminator] Should discriminator be resolved? Default: true
```
Comment (veronicagg) [inline lib/validators/specValidator.js]: was this not an option in the currently committed code?

File: lib/validators/specValidator.js [hunk 3]
```diff
@@ -74,6 +76,12 @@ class SpecValidator {
     if (options.shouldResolvePureObjects === null || options.shouldResolvePureObjects === undefined) {
       options.shouldResolvePureObjects = true;
     }
+    if (options.shouldResolveDiscriminator === null || options.shouldResolveDiscriminator === undefined) {
```
Comment (veronicagg) [inline lib/validators/specValidator.js]: if shouldResolveDiscriminator is passed as "true" in options, should we also make sure to set "shouldResolveAllOf" as true?

**Comment by veronicagg** [comment]
Also, CI is currently failing.

File: lib/validators/specValidator.js (hunk 3 above)
Comment (amarzavery) [reply lib/validators/specValidator.js]: Yup. I am doing that below in lines 81 and 82

Comment (veronicagg) [reply lib/validators/specValidator.js]: Yes, but isn't that happening only if "shouldResolveDiscriminator" is null or undefined? how about if someone set it to true?

File: lib/validators/specValidator.js (hunk 2 above)
Comment (amarzavery) [reply lib/validators/specValidator.js]: I missed documenting it in the specValidator.js class. Actually specValidator simply passes these options to specResolver. So in the new commit, I have removed those options from specValidator as we are checking and validating those in the specResolver cosntructor. Hence it makes no sense to do it twice.

File: ChangeLog.md (hunk 1 above)
Comment (amarzavery) [reply ChangeLog.md]: thanks. changed it.

File: lib/validators/specValidator.js (hunk 3 above)
Comment (amarzavery) [reply lib/validators/specValidator.js]: Got it. Have changed it now. Please check the new commit.

File: lib/validators/specValidator.js [hunk 4]
```diff
@@ -64,23 +64,6 @@ class SpecValidator {
...
-      options.shouldSetAdditionalPropertiesFalse = true;
-    }
-    if (options.shouldResolvePureObjects === null || options.shouldResolvePureObjects === undefined) {
-      options.shouldResolvePureObjects = true;
-    }
-    if (options.shouldResolveDiscriminator === null || options.shouldResolveDiscriminator === undefined) {
-      // Resolving allOf is a neccessary precondition for resolving discriminators. Hence hard setting this to true
-      options.shouldResolveAllOf = true;
-      options.shouldResolveDiscriminator = true;
-    }
```
Comment (amarzavery) [inline lib/validators/specValidator.js]: removing the above conditions as they are being validated in the specResolver constructor. We are not exposing the resolver. It is accessed by the validator. SpecValidator is a pass thru.

File: .vscode/launch.json [hunk 5]
```diff
@@ -71,7 +71,7 @@
       "cwd": "${workspaceRoot}",
       "stopOnEntry": false,
       "args": [
-        "test/liveValidatorTests.js"
+        "test/modelValidatorTests.js"
```
Comment (veronicagg) [inline .vscode/launch.json]: is this intentional, should it be one or the other?

Comment (amarzavery) [reply .vscode/launch.json]: It could be any, doesnt matter.

**Pull request merged**